pkgs/applications/networking/cluster/kubectl-ktop/default.nix

```diff
@@ -0,0 +1,36 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "kubectl-ktop";
+  version = "0.3.5";
+
+  src = fetchFromGitHub {
+    owner = "vladimirvivien";
+    repo = "ktop";
+    rev = "v${version}";
+    sha256 = "sha256-nkIRVt2kqsE9QBYvvHmupohnzH+OBcwWwV16rMePw4I=";
+  };
+
+  vendorSha256 = "sha256-IiAMmHOq69WMT2B1q9ZV2fGDnLr7AbRm1P7ACSde2FE=";
+
+  subPackages = [ "." ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/vladimirvivien/ktop/buildinfo.Version=v${version}"
+    "-X github.com/vladimirvivien/ktop/buildinfo.GitSHA=${src.rev}"
+  ];
+
+  postInstall = ''
+    mv $out/bin/ktop $out/bin/kubectl-ktop
+  '';
+
```

Comment:
```suggestion
```
the upstream docs mention ktop as a binary name

Reply:
`kubectl <subcommand>` auto-discovery checks for binaries named `kubectl-<subcommand>` in the `$PATH`.
E.g. `kubectl ktop` => `kubectl-ktop`.

`nix run`/`lib.getExe` correctly picks `kubectl-ktop` because of the `pname` as well.

So what do you suggest?
A symlink?

Reply:
We already do similar manipulations for other kubectl plugins.

Even though ktop works normally as a single binary, I would probably keep the kubectl-<plugin> naming scheme as @IndeedNotJames suggests.

Reply:
Then we should create a symlink because it is confusing if upstream docs no longer apply.